### PR TITLE
CSHARP-4156: Test against MongoDB quarterly rapid releases

### PR DESCRIPTION
**Pull request by JamesKovacs**
Similar to CSHARP-4155 except adding the quarterly rapid releases to our test matrix.

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -1330,6 +1330,14 @@ axes:
         display_name: "latest"
         variables:
            VERSION: "latest"
+      - id: "rapid"
```
Comment (DmitryLukyanov) [inline evergreen/evergreen.yml:1333]: same as in 4155 :) :
>we have a number of envs where we run only 5.0 servers like csfle-with-mocked-kms-tests-windows or where servers are defined explicitly like in secure-tests-macOS or compressions, we may update them too

**Review by DmitryLukyanov — APPROVED** [review]
Question and then LGTM

File: evergreen/evergreen.yml (hunk 1 above)
Comment (JamesKovacs) [reply evergreen/evergreen.yml:1333]: Rebased on latest CSHARP-4155 and pushed updates.

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Pull request merged**